lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
> Why can't route be a ro attribute and that's it?

Because D2::C::Request isn't a Moo class, and does all the OO itself.  There may be an argument that it *should* be converted into a Moo class, but I'd consider that a separate tech debt issue personally.